### DISCUSSION TITLE
Welcome to AMG8833-ESPHOME Discussions!

### DISCUSSION THREAD

**Discussion by TheRealWaldo** [TheRealWaldo/AMG8833-ESPHOME#8] [Q&A]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Zaogir** [comment]
Hi, there is a way ti send the pixels of the camera to a display in esphome ?
my display is st7789v model ADAFRUIT_RR_280X240

**Reply by TheRealWaldo** [reply]
I haven't seen any projects using ESPHome to do that, but there's a lot online using platformio and arduino.

**Reply by Zaogir** [reply]
I'm trying to get the url of the image sent by the camera so that the display can get it but there is no result because it gives me a permissions error.

Thank you anyway for your time! :D

**Reply by TheRealWaldo** [reply]
There is no image; it's raw sensor data. With this ESPHome project, there is no URL to access the data; it is pushed out to the ESPHome server, which then communicates it to Home Assistant. My project, the Thermal Home Assistant HACS integration, then takes that data and converts it into an image.

Correct me if I'm wrong: It sounds like you want to capture data from the AMG8833, render an image, and then display it on your screen.  

If so, there would be no need for all the extra overhead of all the TCP communication, ESPHome, the server, Home Assistant, and the plugin to do that.

**Reply by UniKyrn** [reply]
Yes, it's not sent as an image, it's a data stream that can be turned into an image.

![Capture](https://github.com/TheRealWaldo/AMG8833-ESPHOME/assets/96962658/142a2c18-9540-4cf7-a7a1-f52634cd534c)

**Reply by Zaogir** [reply]
yes that's what i want to do

**Reply by UniKyrn** [reply]
The source to the Thermal Camera project would show how to decode the data stream.  It looks like a very simply compression system.  I added a few local things to my board and that data stream is chatty as hell. :)
![image](https://github.com/TheRealWaldo/AMG8833-ESPHOME/assets/96962658/175b7b68-6d56-4e27-b95d-677f561632cf)

**Reply by TheRealWaldo** [reply]
It is, but once again, if you're just looking at hooking up a display to the AMG8833, none of this is needed, nor warranted, as it's way too much overhead.

Instead, take the data directly from the sensor, which is simply expressed as values between 1-255, assign them to a spectrum, generate your image, and put it on the display directly in your ESP or ESP32.

Many displays work with simple x,y co-ordinate and color assignment.  It would be a few lines of code in an arduino or platformio project.

**Reply by TheRealWaldo** [reply]
Here's an example: https://learn.adafruit.com/adafruit-amg8833-8x8-thermal-camera-sensor

**Reply by Zaogir** [reply]
I will do it with arduino ide thank you!
